packages/storage/amplify_storage_s3_dart/lib/src/storage_s3_service/service/storage_s3_service_impl.dart

```diff
@@ -562,6 +563,20 @@ class StorageS3Service {
     );
   }
 
+  s3.S3Client _createS3Client({
+    bool useAcceleration = false,
+  }) =>
+      // dependencyManager.get() => S3Client is used for unit tests
+      _dependencyManager.get() ??
+      s3.S3Client(
+        region: _defaultRegion,
+        credentialsProvider: _credentialsProvider,
+        s3ClientConfig: smithy_aws.S3ClientConfig(
+          useAcceleration: useAcceleration,
+        ),
+        client: _httpClient,
```

Comment:
question: Will this create a new httpClient for each s3 client? If so, would it be possible to reuse the HTTP client? I'm not sure about our HTTP client, but the package:http HTTP client has a lot of performance gains from reusing the same client on the same host bc if you create new ones, it has to do a new handshake for each client. Even if our HTTP client doesn't have that issue, might be nice to reuse.

Reply:
Good to know, I think for now it's safe to use the same HTTP client instance, will update.

Reply:
Updated to use a singleton of HTTP client with in the same instance of S3Service wrapper.

Reply:
This does seem pretty inefficient. Seems like we should have a max of 2 S3Clients floating around.

```dart
_dependencyManager.get() ?? (useAcceleration ? _accelerateClient : _normalClient);

late final _accelerateClient = s3.S3Client(...);
late final _normalClient = s3.S3Client(...);
```

Reply:
But actually it looks like this should be configurable on the client methods since it's configurable on the operation currently. So I can put up a PR for that at some point.

Reply:
Thanks for updating the SDK, no it doesn't need to create multiple instances of `S3Client`.